src/Farmer/Common.fs

```diff
@@ -598,9 +598,15 @@ module Sql =
             | PremiumPool c ->
                 c
 
+module ManagedIdentity =
+    /// A user assigned managed identity that can be associated with a resource.
+    type UserAssignedIdentity = UserAssignedIdentity of Name:string * ResourceGroup:string option
+
 module ContainerGroup =
     type PortAccess = PublicPort | InternalPort
     type RestartPolicy = NeverRestart | AlwaysRestart | RestartOnFailure
+    /// Identity settings for a container group.
+    type ContainerGroupIdentity = SystemAssigned | UserAssigned of ManagedIdentity.UserAssignedIdentity list
```

Comment:
I wonder whether this is going to be a common pattern? I assume so, in which case we may wish to consider promoting this type into ManagedIdentity in general.